core/src/main/java/org/elasticsearch/tasks/CancellableTask.java

```diff
@@ -29,9 +29,12 @@
 public class CancellableTask extends Task {
 
     private final AtomicReference<String> reason = new AtomicReference<>();
+    private final boolean shouldCancelChildrenOnCancellation;
 
-    public CancellableTask(long id, String type, String action, String description, TaskId parentTaskId) {
+    public CancellableTask(long id, String type, String action, String description, TaskId parentTaskId,
```

Comment:
After seeing the mess that these kinds of "constants your children set" have caused in aggregations I'd prefer to have `shouldCancelChildrenOnCancellation` be abstract.